packages/cli/test/integration/shared/utils/testCommand.ts

```diff
@@ -0,0 +1,37 @@
+import type { Config } from '@oclif/core';
+import type { Class } from 'n8n-core';
+import { mock } from 'jest-mock-extended';
+
+import type { BaseCommand } from '@/commands/BaseCommand';
+import * as testDb from '../testDb';
+
+export const setupTestCommand = <T extends BaseCommand>(Command: Class<T>) => {
+	const config = mock<Config>();
+	config.runHook.mockResolvedValue({ successes: [], failures: [] });
+
+	// mock SIGINT/SIGTERM registration
+	process.once = jest.fn();
+
+	beforeAll(async () => {
+		await testDb.init();
+	});
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	afterAll(async () => {
+		await testDb.terminate();
```

Comment:
Nit - Some `testDb` calls happen in the suite, some happen here. This dries up tests but might condition us to e.g. forget about `testDb.terminate` in tests that are not about CLI commands and do not have this convenience? Personally I'd prefer not to switch back and forth to see what the suite is doing. Really minor though.